doc/reference/reference_lua/config/utils_schema.rst

```diff
@@ -0,0 +1,1163 @@
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_env-vars`
+
+..  _config-utils-schema-map:
+
+..  function:: schema.map(opts)
+
+    Create a map node of a configuration schema.
+
+    :param table opts: a table in the following format:
+
+                       ..  code-block:: text
+
+                           { key = <schema_node_object>, value = <schema_node_object>, <..annotations..> }
+
+                       See also: :ref:`schema_node_object <config-utils-schema_node_object>`, :ref:`schema_node_annotation <config-utils-schema_node_annotation>`.
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_map`
+
+..  _config-utils-schema-new:
+
+..  function:: schema.new(schema_name, schema_node[, { methods = <...> }])
+
+    Create a schema object.
+
+    :param string schema_name: a name
+    :param table schema_node: a root schema node
+    :param table methods: methods
+
+    :return: a new schema instance (see :ref:`schema_object <config-utils-schema_object>`)
+    :rtype: userdata
+
+    **See also:** :ref:`config_utils_schema_getting_started`
+
+..  _config-utils-schema-record:
+
+..  function:: schema.record(fields[, annotations])
+
+    Create a record node of a configuration schema.
+
+    :param table fields: a table of fields in the following format:
+
+                         ..  code-block:: text
+
+                             { [<field_name>] = <schema_node_object>, <...> }
+
+                         See also: :ref:`schema_node_object <config-utils-schema_node_object>`.
+
+    :param table annotations: annotations (see :ref:`config_utils_schema_annotation`)
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_record`
+
+..  _config-utils-schema-scalar:
+
+..  function:: schema.scalar(type[, annotations])
+
+    Create a scalar node of a configuration schema.
+
+    :param string type: data type (see :ref:`config_utils_schema_data_types`)
+    :param table annotations: annotations (see :ref:`config_utils_schema_annotation`)
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_scalar`
+
+..  _config-utils-schema-set:
+
+..  function:: schema.set(allowed_values, annotations)
+
+    Shortcut for creating an array node of unique string values from the given list of allowed values.
+
+    :param table allowed_values: allowed values of array items
+    :param table annotations: annotations (see :ref:`config_utils_schema_annotation`)
+
+    :return: the created schema node
+    :rtype: table
+
+    **See also:** :ref:`config_utils_schema_nodes_array`
+
+..  _config-utils-schema_object:
+
+schema_object
+~~~~~~~~~~~~~
+
+..  class:: schema_object
+
+    ..  _config-schema_object-apply_default:
+
+    ..  method:: apply_default(data)
```

Comment:
From the documentation request:

> Important: the data is assumed as already validated against the given schema.

And it is actually important -- this is a part of the method's contract. Please, include it into the API reference.

The same remark is applicable for some other methods -- see the documentation request.

Reply:
Added this note where applicable